From 6dcd102b17d1c6c4fa9f743b5346bad66c4cd073 Mon Sep 17 00:00:00 2001
From: Karthik M <quic_karm@quicinc.com>
Date: Tue, 6 Dec 2022 17:34:27 +0530
Subject: [PATCH] ath12k: Fixed MCS rate setting fix for 2G radio

The EHT capability is advertised as part of ieee80211_sta_eht_cap.

We have recently added the check to verify MCS 14 and 15 for all radios.
For 2G radio, MCS 14 capabilty is not supported. We need to handle the mcs
mask accordingly to allow the bitrate setting.

Signed-off-by: Karthik M <quic_karm@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/wireless/ath/ath12k/mac.c b/drivers/net/wireless/ath/ath12k/mac.c
index c550884..37677c7 100644
--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -12015,6 +12015,7 @@ ath12k_mac_eht_mcs_range_present(struct ath12k *ar,
 			break;
 		case BIT(15) - 1:
 		case BIT(16) - 1:
+		case BIT(16) - BIT(14) - 1:
 			if (i != 0)
 				return false;
 			break;
-- 
2.17.1

